lang/src/api.rs

```diff
@@ -69,34 +69,72 @@ pub enum TypeDescription {
     Address,
     /// The SRML balance type.
     Balance,
+    /// The tuple type
+    Tuple {
```

Comment:
I just got the idea that we can even improve this further by doing something similar to how `syn` handles all its complex `enum` types: https://docs.rs/syn/0.15.29/syn/enum.Type.html

So we could define a `TupleTypeDescription` and `ArrayTypeDescription` where all the tuple- and array type description specific implementations are going. They would both implement their own `TryFrom` for `syn::TypeTuple` and `syn::TypeArray` respectively and we could have a much simpler `TryFrom` impl for `TypeDescription`.